go/arrow/scalar/scalar.go

```diff
@@ -311,6 +314,50 @@ func NewDecimal128Scalar(val decimal128.Num, typ arrow.DataType) *Decimal128 {
 	return &Decimal128{scalar{typ, true}, val}
 }
 
+type Decimal256 struct {
+	scalar
+	Value decimal256.Num
+}
+
+func (s *Decimal256) value() interface{} { return s.Value }
+
+func (s *Decimal256) String() string {
+	if !s.Valid {
+		return "null"
+	}
+	val, err := s.CastTo(arrow.BinaryTypes.String)
+	if err != nil {
+		return "..."
+	}
+	return string(val.(*String).Value.Bytes())
+}
+
+func (s *Decimal256) equals(rhs Scalar) bool {
+	return s.Value == rhs.(*Decimal256).Value
+}
+
+func (s *Decimal256) CastTo(to arrow.DataType) (Scalar, error) {
+	if !s.Valid {
+		return MakeNullScalar(to), nil
+	}
+
+	switch to.ID() {
+	case arrow.DECIMAL256:
+		return NewDecimal256Scalar(s.Value, to), nil
```

Comment:
I think this is missing some casting of `s.Value` in case the scales don't match.
And even if the scales match, should probably check that the value still fits in the new precision.

Reply:
fixed this, calling `Rescale` and `FitsInPrecision`